src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChange, SimpleChanges } from '@angular/core'
+import { AbstractControl, FormBuilder, UntypedFormGroup, Validators } from '@angular/forms'
+import { ActionTrigger, CreateActionTrigger, KeyboardAndSelectionTriggerData, KeyboardTriggerData, SHORTCUT_KEYS_MAPPINGS } from 'src/app/shared/models/action-trigger'
+import { Tv2PartAction } from 'src/app/shared/models/tv2-action'
+import { ActionStateService } from 'src/app/shared/services/action-state.service'
+import { Logger } from 'src/app/core/abstractions/logger.service'
+import { ActionTriggerService } from 'src/app/shared/abstractions/action-trigger.service'
+
+export interface ActionTriggerUIForm {
+  actionId: string
+  data: {
+    keys: string
+    actionArguments: number
+  }
+}
+
+@Component({
+  selector: 'sofie-edit-action-triggers',
+  templateUrl: './edit-action-triggers.component.html',
+  styleUrls: ['./edit-action-triggers.component.scss'],
+})
+export class EditActionTriggersComponent implements OnChanges, OnInit {
+  @Output() private readonly cancelActionTrigger: EventEmitter<void> = new EventEmitter<void>()
+  @Input() public selectedActionTrigger: ActionTrigger<KeyboardAndSelectionTriggerData> | null
+  public submitting: boolean = false
+  public submitted: boolean = false
+  public actions: Tv2PartAction[]
+  public loading: boolean = false
+  public keyPress: boolean = false
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly actionTriggerService: ActionTriggerService,
+    private readonly logger: Logger,
+    private readonly actionStateService: ActionStateService
+  ) {}
+
+  public actionForm: UntypedFormGroup = this.fb.group({
+    actionId: ['', [Validators.required]],
+    data: this.fb.group({
+      keys: ['', [Validators.required]],
+      actionArguments: '',
+    }),
+  })
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const actionChange: SimpleChange | undefined = changes['selectedActionTrigger']
+    if (actionChange) {
+      const action: ActionTrigger<KeyboardAndSelectionTriggerData> | null = actionChange.currentValue
+      if (action) {
+        this.patchValue(action)
+      } else {
+        this.actionForm.reset()
+      }
+    }
+  }
+
+  public ngOnInit(): void {
+    this.loading = true
+    this.actionStateService
+      .subscribeToRundownActions('jSXbtcsHTPjebGXurMzP401Z3u0_')
+      .then(observable => observable.subscribe(actions => this.onActionsChanged(actions as Tv2PartAction[])))
+      .catch(error => this.logger.data(error).error('Error while listening to Action events'))
+  }
+
+  private onActionsChanged(loadedActions: Tv2PartAction[]): void {
```

Comment:
Minor thing: The `loaded` part of `loadedActions` refers to where the Actions are coming from _outside_ of the scope of this method. As mentioned earlier methods does not and should not know anything about the outside scope.

There is nothing stopping me from doing something like:
```
public removeAllButFirstFiveActions(): void {
  const changedActions = this.actions.slice(0, 5)
  this.onActionsChanged(changedActions)
}
```
In the above example I modify the Actions and utilize `onActionsChanged()` to ensure everything that should happen when the Actions are changed happens. But in this flow the Actions are never loaded.

The common pitfall for why we often end up with names that goes outside the scope of methods etc is because we have a tendency to name after _where_ we want to use our code instead of naming it after _what_ our code does.

Reply:
I will make deep copy of the object to remove the refers. 
`this.actions = JSON.parse(JSON.stringify(loadedActions))`